File: lib.rpiws281x/src/main/java/com/externalalarmclock/lib/rpiws281x/IRpiWs281x.java
package com.externalalarmclock.lib.rpiws281x;

import java.awt.Color;
import java.util.Collection;
import java.util.List;
import java.util.Map;

public interface IRpiWs281x {
	void init(int dma, int freq, RpiWs281xChannel... channels);

	void render(Map<RpiWs281xChannel, List<Color>> pixels);

	void fini();

	Collection<RpiWs281xChannel> getChannels();
}

File: lib.rpiws281x/src/main/kotlin/com/externalalarmclock/lib/rpiws281x/IRpiWs281x.kt
package com.externalalarmclock.lib.rpiws281x

import java.awt.Color

interface IRpiWs281x {
    val channels: Collection<RpiWs281xChannel>
    fun init(dma: Int, freq: Int, vararg channels: RpiWs281xChannel)

    fun render(pixels: Map<RpiWs281xChannel, List<Color>>)

    fun fini()
}